### PR TITLE
migrate to configuration management

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/18290

## Checklist

- [x] Update CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Wording update: change "control plane" to "management cluster"
 - Make customer connector descriptions more user friendly
+- Prepare helm values to configuration management.
+- Update architect-orb to v3.0.0.
```
Comment (marians) [inline CHANGELOG.md:15]:
```suggestion
- Update architect-orb to v4.0.0.
```

File: helm/dex-app/templates/dex-k8s-authenticator/deployment-customer.yaml [hunk 2]
```diff
@@ -19,8 +19,8 @@ spec:
     spec:
       containers:
       - name: {{ .Chart.Name }}
-        {{ if (.Values.Installation) }}
-        image: "{{ .Values.Installation.V1.Registry.Domain }}/{{ .Values.client.image.name }}:{{ .Values.client.image.tag }}"
+        {{ if .Values.isManagementCluster }}
+        image: "{{ .Values.registry.doomain }}/{{ .Values.client.image.name }}:{{ .Values.client.image.tag }}"
```
Comment (marians) [inline helm/dex-app/templates/dex-k8s-authenticator/deployment-customer.yaml:23]: `doomain` looks fishy

File: helm/dex-app/templates/dex-k8s-authenticator/deployment-customer.yaml [hunk 3]
```diff
@@ -70,7 +70,7 @@ spec:
       - name: config
         configMap:
           name: {{ include "resource.dexk8sauth.name" . }}-customer
-      {{- if not .Values.Installation.V1.GiantSwarm.OIDC.Letsencrypt }}
+      {{- if not .Values.ingress.tls.etsencrypt }}
```
Comment (marians) [inline helm/dex-app/templates/dex-k8s-authenticator/deployment-customer.yaml:73]:
```suggestion
      {{- if not .Values.ingress.tls.letsencrypt }}
```

**Comment by marians** [comment]
Found some typos

**Comment by corest** [comment]
It is weird I've requested review as I don't remember this

**Comment by marians** [comment]
Probably it happened automatically because of CODEOWNERS

image

**Review by marians — APPROVED** [review]

**Pull request merged**